### PR TITLE
implement log following when initiating build runs from builds

### LINKED ISSUE #10: add -F option to 'bd run' so you can directly tail the logs for the build you start
I suggested this to @alicerum during some of her recently merged PRs

I then learned that based on prior planning, that was declared out of scope for her recently completed set of work.

Per conversation with @adambkaplan I'm opening an issue to track this.

Feels to me like a good way for me to tinker with some code, get familiar.

FYI, we have something very analogous to this with openshift builds, namely `oc start-build -F`

### PR DESCRIPTION
**Pull request by gabemontero**
# Changes

Fixes #10 

This is a new PR that stemmed form a week or two of collaboration in #15 

I decided to keep a new PR and close #15 so as to preserve the use of tkn history that is there in case something down the line forces us to switch back to that approach for this particular scenario, but still leverage the pod watching here for @otaviof 's source upload work.

/assign @otaviof 

this is almost entirely a squashed commit of the ones you provided last friday, plus the changes I provided, where there was not much left of my forays when I removed unused alternatives

I did add a new commit to the run.go onEvent switch to use a default clause for pending/unknown and remove the preceding if ... I think this most accurately matches what I found in tkn that we discussed, as well as accounts for pod phases we did not account for, as well as general best practice wrt switch stmts

# Submitter Checklist

- [ ] Includes tests if functionality changed/was added
- [ ] Includes docs if changes are user-facing
- [ ] [Set a kind label on this PR](https://prow.k8s.io/command-help#kind)  
- [ ] Release notes block has been filled in, or marked NONE

See [the contributor guide](https://github.com/shipwright-io/build/blob/main/CONTRIBUTING.md)
for details on coding conventions, github and prow interactions, and the code review process.

# Release Notes

```release-note
A command line argument (-F or --follow) was added to shp build run so that the logs for the resulting build run are piped in real time to stdout and stderr
```

### CODE REVIEW

File: pkg/shp/cmd/build/run.go [hunk 1]
```diff
@@ -19,8 +23,14 @@ import (
...
 	cmd *cobra.Command // cobra command instance
 
+	ioStreams       *genericclioptions.IOStreams // io-streams instance
+	pw              *reactor.PodWatcher          // pod-watcher instance
+	logTail         *tail.Tail                   // follow container logs
+	tailLogsStarted map[string]bool              // controls tail instance per container
+
 	buildName    string                      // build name
 	buildRunSpec *buildv1alpha1.BuildRunSpec // stores command-line flags
+	Follow       bool                        // flag to tail pod logs
```
Comment (otaviof) [inline pkg/shp/cmd/build/run.go:33]: nit: this probably should be a private attribute.

**Review by otaviof — APPROVED** [review]
/lgtm

Thanks!

File: pkg/shp/cmd/build/run.go (hunk 1 above)
Comment (gabemontero) [reply pkg/shp/cmd/build/run.go:33]: fwiw this was not an oversight, but explicit choice based on what they do in `oc`

for example, https://github.com/openshift/oc/blob/master/pkg/cli/startbuild/startbuild.go#L94-L138
https://github.com/openshift/oc/blob/master/pkg/cli/cancelbuild/cancelbuild.go#L63-L77

and a bunch more...admittedly, I did not fully research how this pattern manifested

but we'll see how `shp` evolves, and sure, perhaps we break ranks here and stay consistent in the other direction

**Pull request merged**